Source/Project64/N64 System/Mips/Pif Ram.cpp

```diff
@@ -297,19 +297,16 @@ void CPifRam::SI_DMA_READ()
 			{
 				continue;
 			}
-			RDRAM[RdramPos ^3] = m_PifRam[count];
+			RDRAM[BES(RdramPos)] = m_PifRam[count];
 		}
 	}
 	else
 	{
-		for (size_t i = 0; i < 64; i += 4)
-		{
-			unsigned __int32 pif_ram_dword;
-			std::memcpy(&pif_ram_dword, &PifRamPos[i], sizeof(unsigned __int32));
-
-			pif_ram_dword = swap32by8(pif_ram_dword);
+		register size_t i;

```

Comment:
>There's no particular reason to leave this in an outer scope it's not used in.

That assumes the view that "scope" should exist in between.

If you suppose that in between the body of a loop (such as `for`) and just outside the loop, should be a scope of its own (the control and heading of the loop), then it is your opinion that the declaration should go to the loop header.

In particular C99 and C++ should allow just as you say, but I find that there's also no particular reason for me to assume that this slight increase in readability to humans varies directly with ease of parse-ability (and readability) to compilers.  (The same logical issues with direct language parsing having constituted some of the options held back before C99.)  I actually find that the C89 version looks less open to interpretation and less ambiguous, IMO, and easier for me to read in my honest opinion, if not also easier for the compiler to parse faster.

But if it looks just that much harder for you or @project64 to read, I will change it back.

> By the way, have you benchmarked or checked that the assembly produced is actually better with the register keyword.  I can bet it does not alter semantics at all

Then you propose that my purpose for specifying `register` is the pretense that it necessarily makes the generated code any different or run faster?

Like any hint for any language, I'm aware that it makes no automatic difference in the end result.  It is just a compiler hint--one which often goes ignored (especially in this case, as the compiler's job is to optimize which should be registers, which should be on the stack, and which should be in-lined or even exist at all).  (And it **usually** has all the information it needs to correctly decide such, for most basic algorithms.)

But unless you can find a reason to suggest that there's anything to lose from my approach, and not just that "there's almost never anything to gain", then I'm really not sure what positive difference it makes to change that.